phantom-dsl/src/main/scala/com/outworkers/phantom/macros/TableHelper.scala

```diff
@@ -390,56 +385,44 @@ class TableHelperMacro(override val c: whitebox.Context) extends RootMacro(c) {
 
   def macroImpl[T : WeakTypeTag, R : WeakTypeTag]: Tree = {
     val tableType = weakTypeOf[T]
-    val rTpe = weakTypeOf[R]
+    val recordType = weakTypeOf[R]
     val refTable = determineReferenceTable(tableType).map(_.typeSignature).getOrElse(tableType)
     val referenceColumns = refTable.decls.sorted.filter(_.typeSignature <:< typeOf[AbstractColumn[_]])
     val tableName = extractTableName(refTable)
-
     val columns = filterMembers[T, AbstractColumn[_]](exclusions)
-    val descriptor = extractor(tableType, rTpe, referenceColumns)
-    val abstractFromRow = refTable.member(fromRowName)
-    val fromRowTpe = abstractFromRow.infoIn(tableType)
-    val fromRowFn = descriptor.fromRow
-
-    if (fromRowFn.isEmpty) {
-      logger.debug(
-        s"""
-          Table: ${printType(tableType)}
-          Type info: ${printType(fromRowTpe)}
-          fromRowDefined: ${descriptor.fromRow.isDefined}
-          fromRow == ???: ${abstractFromRow.asMethod}
-          abstract: ${abstractFromRow.asMethod.isAbstract}
-          abstractOverride: ${abstractFromRow.asMethod.isAbstractOverride}
-          body: ${showCode(q"$fromRowTpe")}
-        """
-      )
-    } else {
-      logger.debug(descriptor.showExtractor)
-    }
-
+    val descriptor = extractor(tableType, recordType, referenceColumns)
     val accessors = columns.map(_.asTerm.name).map(tm => q"table.instance.${tm.toTermName}").distinct
     val clsName = TypeName(c.freshName("anon$"))
+    val notImpemented = q"???"
 
     q"""
-       final class $clsName extends $macroPkg.TableHelper[$tableType, $rTpe] {
+       final class $clsName extends $macroPkg.TableHelper[$tableType, $recordType] {
           type Repr = ${descriptor.storeType}
```

Comment:
@iRhysBradburyOTRL It's not particularly straightforward stuff, and it's still pretty quirky in the sense that the compiler is not yet smart enough to see the generated type. I don't really know how to make it see the type, but I'll fix that.